src/runtime/components/LitWrapperServer.ts

```diff
@@ -39,17 +40,17 @@ export default defineComponent({
       return Promise.all(childToHtmlPromises);
     },
 
-    attachPropsToRenderer() {
+    attachPropsToRenderer(): void {
       const customElementConstructor = getCustomElementConstructor(this.litElementTagName);
       const props = this.litElementVnode.props;
 
-      if (props) {
+      if (props && this.renderer) {
         for (const [key, value] of Object.entries(props)) {
           // check if this is a reactive property
           if (
-            customElementConstructor !== null &&
-            typeof customElementConstructor !== "string" &&
-            key in customElementConstructor.prototype
+            customElementConstructor !== null && // The element exists
+            typeof customElementConstructor !== "string" && // It's not just a string
+            key in customElementConstructor.prototype // The property we're looking for is on the prototype
           ) {
             this.renderer.setProperty(key, value);
```

Comment:
We need to handle key-only boolean props here specifically.

E.g. `<accordion-item open>...</accordion-item>`

Because Vue will give the value of the boolean prop as an empty string, so on the server side, such prop won't be rendered correctly. Because LitElementRenderer gets `''` as the prop's value.

Reply:
```JS
const isBooleanProp = customElementConstructor.properties[key]?.type === Boolean;

if (isBooleanProp && value === "") {
  // handle key only boolean props e.g. <my-element disabled></my-element>
  this.renderer.setProperty(key, true);
} else {
  this.renderer.setProperty(key, value);
}
  ```

Reply:
The syntax as it is in this PR seems to handle boolean properties correctly, producing `<accordion-item title="foo" open>` for the open one and `<accordian-item title="bar">` for the closed one

Reply:
What I noticed is that the `open` attribute is set. However, in the Lit element itself, the value of the `open` prop is `""`. So the component does not actually open since it does checks like `?hidden=${!this.open}`

Reply:
Was the `open` property made on the element as `{type: Boolean}`? I think it should be able to handle empty string as value if so.

Reply:
@augustjk yes, it's set as a boolean, but it doesn't convert `""` into `true` when we call `setProperty` - I think hydration can cope with it but not the element renderer.

We may have to make a special case for boolean values here, or it doesn't server-render them correctly

Reply:
I just checked and the default converter is only run when `setAttribute` is called, not `setProperty`. :( I guess the custom handling of boolean is necessary as Vue doesn't do that conversion for us.

Reply:
We can add that as a temporary solution here. But I think it makes sense to do the default conversion also for a property, right?

Reply:
I think the special case handling might need to stay permanently. The converters in Lit are meant for converting attributes to property when users provide them as part of the HTML markup. In Lit or vanilla HTML/JS, we don't want to run a converter when things are set as property as these should already be in the data type we want. With frameworks like Vue and React, that's a bit iffy with everything grouped together. Vue taking a boolean attribute/prop and setting value as empty string in the props bag is a specific problem, that this workaround will always be necessary.